### PR TITLE
add activate API

### PR DESCRIPTION
**Pull request by yosriz**
changes trust for account, to trust Kin asset

some minor refactoring for a common code

### CODE REVIEW

File: kin-sdk-core/src/main/java/kin/sdk/core/KinAccount.java [hunk 1]
```diff
@@ -69,4 +69,24 @@ TransactionId sendTransactionSync(String publicAddress, String passphrase, BigDe
...
+     * Create {@link Request} for activating this account, should be done after account was created
+     * <p> See {@link KinAccount#activateSync(String)} for possibles errors</p>
+     *
+     * @param passphrase the passphrase used to create the account
+     * @return {@code Request<Void>}
+     */
+    Request<Void> activate(String passphrase);
+
+    /**
+     * Activate this account, should be done after account was created
```
Comment (berryve) [inline kin-sdk-core/src/main/java/kin/sdk/core/KinAccount.java]: calling the client createAccount method is not enough for the account to be created
after the KinAccount object that contains KeyPair has been created we need to request from server to actually create the account and fund it with lumens. The documentation here is confusing

File: kin-sdk-core/src/main/java/kin/sdk/core/AccountActivator.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    AccountActivator(Server server, KeyStore keyStore, KinAsset kinAsset) {
+        this.server = server;
+        this.kinAsset = kinAsset;
+        this.keyStore = keyStore;
+    }
+
+    void activate(Account account, String passphrase) throws OperationFailedException {
+        AccountResponse accountResponse;
+        try {
+            accountResponse = server.accounts().account(KeyPair.fromAccountId(account.getAccountId()));
```
Comment (berryve) [inline kin-sdk-core/src/main/java/kin/sdk/core/AccountActivator.java]: check for null

File: kin-sdk-core/src/main/java/kin/sdk/core/AccountActivator.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+            if (kinAsset.hasKinTrust(accountResponse)) {
+                return;
+            }
+            SubmitTransactionResponse response = sendAllowKinTrustOperation(account, passphrase, accountResponse);
+            if (!response.isSuccess()) {
```
Comment (berryve) [inline kin-sdk-core/src/main/java/kin/sdk/core/AccountActivator.java]: response null ?

File: kin-sdk-core/src/main/java/kin/sdk/core/AccountActivator.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+import org.stellar.sdk.responses.AccountResponse;
+import org.stellar.sdk.responses.HttpResponseException;
+import org.stellar.sdk.responses.SubmitTransactionResponse;
+
+class AccountActivator {
+
+    //unlimited trust, The largest amount unit possible in Stellar
+    //see https://www.stellar.org/developers/guides/concepts/assets.html
+    private static final String TRUST_NO_LIMIT_VALUE = "922337203685.4775807";
+    private final Server server;
```
Comment (berryve) [inline kin-sdk-core/src/main/java/kin/sdk/core/AccountActivator.java]: // horizon server

File: kin-sdk-core/src/main/java/kin/sdk/core/Utils.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+import org.stellar.sdk.responses.SubmitTransactionResponse;
+import org.stellar.sdk.responses.SubmitTransactionResponse.Extras.ResultCodes;
+
+final class Utils {
+
+    private Utils() {
+
+    }
+
+    static TransactionFailedException createTransactionException(SubmitTransactionResponse response)
```
Comment (berryve) [inline kin-sdk-core/src/main/java/kin/sdk/core/Utils.java]: @NonNull

**Review by berryve — APPROVED** [review]
fix comments ...

File: kin-sdk-core/src/main/java/kin/sdk/core/AccountActivator.java [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+import org.stellar.sdk.Transaction;
+import org.stellar.sdk.responses.AccountResponse;
+import org.stellar.sdk.responses.HttpResponseException;
+import org.stellar.sdk.responses.SubmitTransactionResponse;
+
+class AccountActivator {
+
+    //unlimited trust, The largest amount unit possible in Stellar
+    //see https://www.stellar.org/developers/guides/concepts/assets.html
+    private static final String TRUST_NO_LIMIT_VALUE = "922337203685.4775807";
```
Comment (YohayBar) [inline kin-sdk-core/src/main/java/kin/sdk/core/AccountActivator.java:20]: do we need this value? I think that if no value provided it use stellar default anyway

Comment (yosriz) [reply kin-sdk-core/src/main/java/kin/sdk/core/AccountActivator.java:20]: it doesn't, it's just how the Stellar laboratory expose this, if you look at the generated XDR this value will be generated.
the same goes with the Java SDK, I cannot send empty string, I must provide value myself.

Comment (YohayBar) [reply kin-sdk-core/src/main/java/kin/sdk/core/AccountActivator.java:20]: cool

**Pull request merged**